src/sdkClient/clientAttributesDecoration.ts

```diff
@@ -0,0 +1,119 @@
+import { AttributesCacheInMemory } from '../storages/inMemory/AttributesCacheInMemory';
+import { validateAttributesDeep } from '../utils/inputValidation/attributes';
+import { SplitIO } from '../types';
+import { ILogger } from '../logger/types';
+import { objectAssign } from '../utils/lang/objectAssign';
+
+/**
+ * Add in memory attributes storage methods and combine them with any attribute received from the getTreatment/s call 
+ */
+export function ClientAttributesDecoration<TClient extends SplitIO.IClient | SplitIO.IAsyncClient>(log: ILogger, client: TClient): any {
```

Comment:
Rename to lower case: `clientAttributesDecoration`. For convention, we use lowercase for factory functions (E.g.: `= myFactory()`) and uppercase for classes -a.k.a constructor functions- (E.g., `= new MyClass()`)